astropy/coordinates/tests/test_sky_coord_velocities.py

```diff
@@ -0,0 +1,199 @@
+
+
+def test_creation_copy_basic():
+    i = ICRS(1*u.deg, 2*u.deg, pm_ra_cosdec=.2*u.mas/u.yr, pm_dec=.1*u.mas/u.yr)
+    sc = SkyCoord(i)
+    sc_cpy = SkyCoord(sc)
+
+    for attrnm in ['ra', 'dec', 'pm_ra_cosdec', 'pm_dec']:
+        assert_quantity_allclose(getattr(sc, attrnm), getattr(sc_cpy, attrnm))
+
+
+def test_creation_copy_rediff():
+    sc = SkyCoord(1*u.deg, 2*u.deg,
+                  pm_ra=.2*u.mas/u.yr, pm_dec=.1*u.mas/u.yr,
+                  differential_cls=SphericalDifferential)
+
+    sc_cpy = SkyCoord(sc)
+    for attrnm in ['ra', 'dec', 'pm_ra', 'pm_dec']:
+        assert_quantity_allclose(getattr(sc, attrnm), getattr(sc_cpy, attrnm))
+
+    sc_newdiff = SkyCoord(sc, differential_cls=SphericalCosLatDifferential)
+    reprepr = sc.represent_as(SphericalRepresentation, SphericalCosLatDifferential)
+    assert_quantity_allclose(sc_newdiff.pm_ra_cosdec,
+                             reprepr.differentials['s'].d_lon_coslat)
+
+
+def test_creation_cartesian():
+    rep = CartesianRepresentation([10, 0., 0.]*u.pc)
+    dif = CartesianDifferential([0, 100, 0.]*u.pc/u.Myr)
+    rep = rep.with_differentials(dif)
+    c = SkyCoord(rep)
+
+    sdif = dif.represent_as(SphericalCosLatDifferential, rep)
+    assert_quantity_allclose(c.pm_ra_cosdec, sdif.d_lon_coslat)
+
+
+def test_useful_error_missing():
+    sc_nod = SkyCoord(ICRS(1*u.deg, 2*u.deg))
+    try:
+        sc_nod.l
+    except AttributeError as e:
+        # this is double-checking the *normal* behavior
+        msg_l = e.args[0]
+
+    try:
+        sc_nod.pm_dec
+    except Exception as e:
+        msg_pm_dec = e.args[0]
+
+    assert "has no attribute" in msg_l
+    assert "has no associated differentials" in msg_pm_dec
+
+
+# ----------------------Operations on SkyCoords w/ velocities-------------------
+
+# define some fixtires to get baseline coordinates to try operations with
+@pytest.fixture(scope="module", params=[(False, False),
+                                        (True, False),
+                                        (False, True),
+                                        (True, True)])
+def sc(request):
+    incldist, inclrv = request.param
+
+    args = [1*u.deg, 2*u.deg]
+    kwargs = dict(pm_dec=1*u.mas/u.yr, pm_ra_cosdec=2*u.mas/u.yr)
+    if incldist:
+        kwargs['distance'] = 213.4*u.pc
+    if inclrv:
+        kwargs['radial_velocity'] = 61*u.km/u.s
+
+    return SkyCoord(*args, **kwargs)
+
+@pytest.fixture(scope="module")
+def sc2():
+    return SkyCoord(1*u.deg, 2*u.deg,
+                pm_dec=1*u.mas/u.yr, pm_ra_cosdec=2*u.mas/u.yr)
+
+@pytest.fixture(scope="module")
+def scmany():
+    return SkyCoord(ICRS(ra=[1]*100*u.deg, dec=[2]*100*u.deg,
+                     pm_ra_cosdec=np.random.randn(100)*u.mas/u.yr,
+                     pm_dec=np.random.randn(100)*u.mas/u.yr,))
+
+
+def test_separation(sc, sc2):
+    sc.separation(sc2)
+
+
+def test_accessors(sc, scmany):
```

Comment:
Does `sc.velocity` work? Now that #6869 is merged, I think we want that to work on skycoord as well.

Reply:
Turns out this reveals a subtle (but probably not *that* important) bug (which is related to the one I referenced above).  I'm just xfailing the case for now.